### PR TITLE
React context for model input data

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

This creates a shared datastore for both the input forms and the curve charts, initialized to match all the input fields in our latest mocks. We are probably going to want to add some more actions later (at the least, probably one to populate the store with state & county level data from an external data source, once we have one set up) but this should get us started for now.

Also I know some TypeScript now? 😎 

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/impact-dashboard/EpidemicModelContext.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+  return context;
+}
+
+function useEpidemicModelDispatch() {
+  const context = React.useContext(EpidemicModelDispatchContext);
+
+  if (context === undefined) {
+    throw new Error(
+      "useEpidemicModelDispatch must be used within a EpidemicModelProvider",
```
Comment (jessex) [inline src/impact-dashboard/EpidemicModelContext.tsx:79]: Nit: within an EpidemicModelProvider

File: src/impact-dashboard/EpidemicModelContext.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import React from "react";
+
+const {
+  populationAndHospitalData,
+} = require("../page-overview/assets/dataSource") as any;
```
Comment (szhu) [inline src/impact-dashboard/EpidemicModelContext.tsx:5]: - Can you move the `dataSource` file up into a more common folder? I'm thinking into `infection-model`. Or can make another top-level folder called `data` or something like that.
- You can rename the file to `.tsx`. You won't have to change anything about the contents of the file, but also you'll now be able to `import` it and not use `any`. 

We don't have to do this in this PR, but I feel like it wouldn't be too hard for a change to make.

File: src/impact-dashboard/ImpactDashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  border-right: 1px solid black;
+  flex: 1 0 auto;
+  width: 350px;
+`;
+
+const ChartsContainer = styled.div`
+  flex: 2 0 auto;
+  width: 350px;
+  margin: 0 15px;
+`;
```
Comment (szhu) [inline src/impact-dashboard/ImpactDashboard.tsx:18]: Thank you for using flexbox! I think this is the easier and more maintainable responsiveness solution for components like this.

File: src/impact-dashboard/EpidemicModelContext.tsx (hunk 2 above)
Comment (macfarlandian) [reply src/impact-dashboard/EpidemicModelContext.tsx:5]: I agree but we need to do a whole new pass at the external datasources to pull in more stuff for this model so I left it alone for now. would rather keep this PR small

File: src/impact-dashboard/EpidemicModelContext.tsx (hunk 1 above)
Comment (macfarlandian) [reply src/impact-dashboard/EpidemicModelContext.tsx:79]: 🤓

**Pull request merged**